### ISSUE TITLE
'one-var' doesn't respect 'separateRequires' option with 'always' or 'consecutive'

### ISSUE THREAD

**Issue by domosapien** [eslint/eslint#10179]
**Tell us about your environment**

* **ESLint Version:** 4.19.1
* **Node Version:** 8.11.1
* **npm Version:** 5.8.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
	'root': true,
	'extends': [],
	'parserOptions': {
		'ecmaVersion': 8
	},
	'env': {
		'es6': true,
		'node': true
	},
	'rules': {
		'no-mixed-requires': [2, { 'allowCall': false }],
		'one-var': [2, { 'var': 'consecutive', 'separateRequires': true }],
	}
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**
Took code from https://eslint.org/docs/rules/one-var#var-let-and-const under the `separateRequires` block.

```js
var foo = require('foo');
var bar = 'bar';

var foo = require('foo'),
	bar = require('bar');
```

Ran through Webstorm, which I presume just runs the configured package (which is the correct path) against the current file.

**What did you expect to happen?**
No errors are flagged.

**What actually happened? Please include the actual, raw output from ESLint.**
2 errors are flagged: The first `bar` variable wants to be joined with the first `foo`. The second foo/bar block wants to be joined with the first block. Interestingly if you make the `consecutive` option an `always`, the first error goes away.

**Comment by rjgotten** [comment]
Same issue. Quite annoying.

Anyone have a known working version that we can downgrade to?

**Comment by platinumazure** [comment]
Looks like this slipped through the cracks. Sorry about that.

@eslint/eslint-team Any thoughts on this issue? Seems like a potential bug to me.

**Comment by ilyavolodin** [comment]
Agree. This looks like a bug. Easier example to reproduce:
```js
/* eslint one-var: [2, { separateRequires: true, var: "always" }] */
var bar = 'bar';

var foo = require('foo');
```

**Comment by ilyavolodin** [comment]
@sstern6 This bug report is accepted, so feel free to work on this whenever you get a chance! Thanks for the help

**Comment by sstern6** [comment]
So looking into this further, I agree the example from @ilyavolodin should not throw an error, however, I think the docs might be wrong.  Before digging in deeper id like to confirm though.  If we have:

```
/* eslint one-var: [2, { separateRequires: true, var: "always" }] */
var foo = require('foo');
var bar = 'bar';

var foo = require('foo'),
	bar = require('bar');
```

Shouldnt this throw an error to join the requires?  Or because they are in separate blocks this example should not throw an error?

**Comment by sstern6** [comment]
Just to clarify this issue, @ilyavolodin example for a failing case is correct.  Worked with @not-an-aardvark to determine the original example for the bug report is a little misleading as is the documentation and should be 2 separate test cases not 1, which currently does not throw any errors.

```
var foo = require('foo');
var bar = 'bar';
```

Should be run independent of:
```
var foo = require('foo'),
	bar = require('bar');
```

Looked into the issue last night and think I understand where the issue is being caused, should have a pr soon.

Thanks

**Comment by sstern6** [comment]
@kaicataldo theres a fix open for this, why did this get closed?

**Comment by Aladdin-ADD** [comment]
I‘ve self-assigned, to avoid the bot closing.

**Comment by korneliakobiela** [comment]
Hi, I can see it is done and added to eslint@5.14.0.  I've updated eslint to this version and tried to add rules: 
``` json
{
    "prefer-const": "error",
    "one-var": [
            "error", 
            {
                "separateRequires": true,
                "const": "consecutive"
            }
        ]
}
```
And then the fact that const is CommonJS require is ignored and eslint shows me an error in that case. What am I doing wrong?
